code/datums/status_effects/xeno_buffs.dm

```diff
@@ -871,3 +871,499 @@
+	return TRUE
+
+/datum/status_effect/upgrade_toxin/on_remove()
+	UnregisterSignal(SSdcs, COMSIG_UPGRADE_CHAMBER_UTILITY)
+	UnregisterSignal(buff_owner, COMSIG_XENOMORPH_ATTACK_LIVING)
+	return ..()
+
+/// Sets the chamber_scaling to the amount of active veil chambers.
+/datum/status_effect/upgrade_toxin/proc/update_buff()
+	SIGNAL_HANDLER
+	chamber_scaling = length(buff_owner.hive.veil_chambers)
+
+/// Injects the human target with a variable amount of the selected reagent.
+/datum/status_effect/upgrade_toxin/proc/on_slash(datum/source, mob/living/target)
+	SIGNAL_HANDLER
+	if(!chamber_scaling || target.stat == DEAD || !ishuman(target) || !target.can_sting())
+		return
+	var/mob/living/carbon/human/human_target = target
+	human_target.reagents.add_reagent(selected_reagent, toxin_amount_per_chamber * chamber_scaling)
+
+// ***************************************
+// ***************************************
+// ***************************************
+/atom/movable/screen/alert/status_effect/upgrade_pheromones
+	name = "Pheromones"
+	desc = "Allows to emit pheromones."
+	icon_state = "xenobuff_phero"
+
+/atom/movable/screen/alert/status_effect/upgrade_pheromones/Click()
+	var/datum/status_effect/upgrade_pheromones/status_effect = attached_effect
+	var/phero_choice = show_radial_menu(status_effect.buff_owner, status_effect.buff_owner, GLOB.pheromone_images_list, radius = 35, require_near = TRUE)
+	if(!phero_choice)
+		return
+	QDEL_NULL(status_effect.current_aura)
+	status_effect.emitted_aura = phero_choice
+	status_effect.create_aura()
+
+/datum/status_effect/upgrade_pheromones
+	id = "upgrade_pheromones"
+	duration = -1
+	status_type = STATUS_EFFECT_UNIQUE
+	alert_type = /atom/movable/screen/alert/status_effect/upgrade_pheromones
+	/// Owner typed as an xenomorph.
+	var/mob/living/carbon/xenomorph/buff_owner
+	/// The aura.
+	var/datum/aura_bearer/current_aura
+	/// The aura to emit.
+	var/emitted_aura = AURA_XENO_RECOVERY
+	/// The initial value of the aura's power.
+	var/aura_power_base = 1
+	/// The phero power to increase by.
+	var/aura_power_per_chamber = 1
+	/// The amount of times to multiply the aura power by.
+	var/chamber_scaling = 0
+
+/datum/status_effect/upgrade_pheromones/on_apply()
+	if(!isxeno(owner))
+		return FALSE
+	buff_owner = owner
+	chamber_scaling = length(buff_owner.hive.veil_chambers)
+	RegisterSignal(SSdcs, COMSIG_UPGRADE_CHAMBER_UTILITY, PROC_REF(update_buff))
+	create_aura()
+	return TRUE
+
+/datum/status_effect/upgrade_pheromones/on_remove()
+	UnregisterSignal(SSdcs, COMSIG_UPGRADE_CHAMBER_UTILITY)
+	if(current_aura)
+		current_aura.stop_emitting()
+	return ..()
+
+/// (Re)creates the aura.
+/datum/status_effect/upgrade_pheromones/proc/create_aura()
+	if(current_aura)
+		QDEL_NULL(current_aura)
+	current_aura = SSaura.add_emitter(buff_owner, emitted_aura, 6 + aura_power_per_chamber * chamber_scaling * 2, aura_power_base + aura_power_per_chamber * chamber_scaling, -1, FACTION_XENO, buff_owner.hivenumber)
```

Comment:
How does this affect xenos who already have pheremones? It seems like this just overrides it, might be good if they buffed it instead

Reply:
They're separate. Pictured normal pheromones as regeneration and mutation pheromones as frenzy:
![image](https://github.com/user-attachments/assets/1e6b35fd-30ec-4b94-9145-f1a28b17a31a)